src/api.js

```diff
@@ -547,21 +547,35 @@ api.declare({
   name:       'createTask',
   stability:  API.stability.stable,
   idempotent: true,
-  scopes:     [
-    // Legacy scope options to be removed, hence, no-longer documented
-    // [
-    //   'queue:create-task:<provisionerId>/<workerType>',
-    // ], [
-    //   'queue:define-task:<provisionerId>/<workerType>',
-    //   'queue:task-group-id:<schedulerId>/<taskGroupId>',
-    //   'queue:schedule-task:<schedulerId>/<taskGroupId>/<taskId>',
-    // ],
-    [
-      'queue:create-task:<priority>:<provisionerId>/<workerType>',
-      'queue:scheduler-id:<schedulerId>',
-    ],
-  ],
-  deferAuth:  true,
+  scopes: {AllOf: [
```

Comment:
Consider refactoring this to be:
`AnyOf: [ {AllOf: [current_scopes]}, {AllOf: [legacy_scopes]}]`
I think that'll read easier..

Reply:
I'm tempted to suggest supporting comments in scope expressions, just to see the look on @imbstack's face.  But seriously, I like this arrangement since it makes it clear what changed.  What if the boolean condition was called `legacyRequestWithoutPriority` or something to suggest this is not preferred?

Reply:
Accurate rendering of the look on my face: 🤕

I do like the idea of rearranging and changing the name of the boolean though.

Reply:
To clarify: I like the arrangement in the PR commits over the arrangement Jonas proposed.